java/src/main/java/ai/rapids/cudf/ArrowColumnBuilder.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *
+ *  Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+
+/**
+ * Column builder from Arrow data. This builder takes in byte buffers referencing
+ * Arrow data and allows efficient building of CUDF ColumnVectors from that Arrow data.
+ * The caller can add multiple batches where each batch corresponds to Arrow data
+ * and those batches get concatenated together after being converted to CUDF
+ * ColumnVectors.
+ * This currently only supports primitive types and Strings, Decimals and nested types
+ * such as list and struct are not supported.
+ */
+public final class ArrowColumnBuilder implements AutoCloseable {
+    private DType type;
+    private final ArrayList<ByteBuffer> data = new ArrayList<>();
+    private final ArrayList<ByteBuffer> validity = new ArrayList<>();
+    private final ArrayList<ByteBuffer> offsets = new ArrayList<>();
+    private final ArrayList<Long> nullCount = new ArrayList<>();
+    private final ArrayList<Long> rows = new ArrayList<>();
+
+    public ArrowColumnBuilder(HostColumnVector.DataType type) {
+      this.type = type.getType();
+    }
+
+    /**
+     * Add an Arrow buffer. This api allows you to add multiple if you want them
```

Comment:
```suggestion
     * Add an Arrow buffer. This API allows you to add multiple if you want them
```

Or alternatively I think it also works without "API" mentioned at all.